Return the proper number of bytes copied when reading box namespaces

Patater opened the pull request:
@meriac 
@AlessandroA

Patater commented:
- Explained why the halt
- Used a + 1 instead of a ++ because _style_

AlessandroA commented on core/system/src/mpu/vmpu.c:
```diff
@@ -449,11 +449,15 @@ static int copy_box_namespace(const char *src, char *dst)
+     * in vmpu_box_namespace_from_id as being in the box config table. It is a
+     * programmer error if the namespace in the box config table is not
+     * null-terminated, so we halt. */
+    HALT_ERROR(SANITY_CHECK_FAILED, "box namespace missing terminating-null");
```
I think the comment got lost with a rebase. Could you capitalize and punctuate the halt error message?

Patater commented:
- Capitalize the halt message. (I didn't add a period because the message is just a sentence fragment.)

AlessandroA commented:
Thanks for the edits! LGTM.

Patater commented:
XXX Don't merge yet.

Patater commented:
- Added commit to mark halt functions as `noreturn`, to get rid of a compile error

Patater commented:
OK to merge if LGTY

AlessandroA commented:
Could you use `UVISOR_NORETURN` for that? (also, please keep the function signature all in one line)

Patater commented:
What is our line limit? And if one function declaration goes over that limit, would you rather the functions be consistently on two line or that only the ones that are too long are on two lines?

Patater commented:
It appears now that many halt functions can return...

This might break some code.

Patater commented:
GitHub is showing the commits weirdly again. Look at the real branch with git to see what is going on, if confused.

- Dropped the UVISOR_NORETURN commit, because halt can maybe return (in the future)
- Rebased on origin/master

Patater commented:
I am concerned if we push the merge button that GitHub will pull in bad things. Discarding this PR in favor of a new one.
